modules/configuration.nix

```
@@ -0,0 +1,183 @@
+    appendConfig = ''
+      error_log /var/log/nginx/error.log debug;
+    '';
+    virtualHosts = {
+      "ymstnt.com" = {
+        enableACME = true;
+        forceSSL = true;
+        root = "/var/www";
+        extraConfig = ''
+          error_page 404 /ymstnt.com-generated/404.html;
+          client_max_body_size 50G;
+          fastcgi_read_timeout 24h;
+        '';
+        locations = {
+          "~ ^([^.\?]*[^/])$".extraConfig = ''
+            if (-d $document_root/ymstnt.com-generated$uri) {
+              rewrite ^([^.]*[^/])$ $1/ permanent;
+            }
+            if (-d $document_root/ymstnt.com$uri) {
+              rewrite ^([^.]*[^/])$ $1/ permanent;
+            }
+            try_files _ @entry;
+          '';
+          "/".extraConfig = ''
+            try_files _ @entry;
+          '';
+          "@entry".extraConfig = ''
+            try_files /ymstnt.com-generated$uri /ymstnt.com-generated$uri/index.html @ymstnt.com-rewrite;
+          '';
+          "@ymstnt.com-rewrite".extraConfig = ''
+            if (-f $document_root/ymstnt.com$uri) {
+              rewrite ^(.*)$ /ymstnt.com$1 last;
+            }
+            if (-f $document_root/ymstnt.com$uri/index.html) {
+              rewrite ^(.*)$ /ymstnt.com$1/index.html last;
+            }
+            if (-f $document_root/ymstnt.com$uri/index.php) {
+              rewrite ^(.*)$ /ymstnt.com$1/index.php last;
+            }
+          '';
+          "/ymstnt.com/".extraConfig = ''
+            alias /var/www/ymstnt.com/;
+            location ~ \.(php|html)$ {
+              alias /var/www;
+              fastcgi_pass unix:${config.services.phpfpm.pools.shared.socket};
+            }
+          '';
+          "/\.git".extraConfig = ''
+            deny all;
+          '';
+          "/.well-known/webfinger".extraConfig = ''
+            rewrite ^.*$ https://social.ymstnt.com/.well-known/webfinger permanent;
+          '';
+          "/.well-known/host-meta".extraConfig = ''
+            rewrite ^.*$ https://social.ymstnt.com/.well-known/host-meta permanent;
+          '';
+          "/.well-known/nodeinfo".extraConfig = ''
+            rewrite ^.*$ https://social.ymstnt.com/.well-known/nodeinfo permanent;
+          '';
+        };
+      };
+    };
+  };
+
+  security.acme = {
+    acceptTerms = true;
+    defaults.email = "ymstnt@mailbox.org";
+  };
+
+  networking.firewall.allowedTCPPorts = [ 80 443 ];
+
+  services.openssh = {
+    enable = true;
+    ports = [ 42727 ];
+    settings = {
+      PasswordAuthentication = false;
+      KbdInteractiveAuthentication = false;
+      PermitRootLogin = "no";
+      X11Forwarding = true;
+    };
+  };
+
+  environment.shellInit = "umask 002";
+  users.users = {
+    shared = {
+      isSystemUser = true;
+      group = "shared";
+    };
+  };
+
+  users.groups.shared = { };
+
+  environment.systemPackages = with pkgs; [
+    git
+    inotify-tools
+    ncdu
+    nh
+    nix-inspect
+    nix-output-monitor
+    nvd
```

Can I move these to `ymstnt` module?

```suggestion
    git
    inotify-tools
    ncdu
    nh
    nix-inspect
    nix-output-monitor
    nvd
```

---

Sure